tests/test_spatial_analysis/test_spatial_interaction.py

```diff
@@ -285,6 +360,56 @@ def test_new_ax_has_right_titles(self):
                     error_msg
                 )
 
+    def test_sinlge_stratify_by(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_num",
+            "Neighborhood Enrichment",
+            stratify_by="cluster_str"
+        )
+        unique_cluster_str_values = self.adata.obs["cluster_str"].unique()
+
+        # Get the keys (unique cluster values) from the dictionary
+        keys = list(ax_dict.keys())
+
+        # Assert that we have at least two keys (clusters)
+        self.assertEqual(len(keys), 2)
+
+        # Assert that the axes associated
+        # with the first and second keys are different
+        self.assertNotEqual(ax_dict[keys[0]], ax_dict[keys[1]])
+
+        for value in unique_cluster_str_values:
+            # Expect each unique value as a key in the returned dict
+            self.assertIn(value, ax_dict.keys())
+
+            # Each should be a matplotlib axis object
+            self.assertIsInstance(ax_dict[value], plt.Axes)
+
+    def test_List_stratify_by(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_str",
+            "Neighborhood Enrichment",
+            stratify_by=[
+                "cluster_num",
+                "labels"
+                ]
+            )
+        combined_keys = self.adata.obs[
+            [
+                "cluster_num",
+                "labels"
+            ]
+             ].astype(str).agg('_'.join, axis=1).unique()
+
+        for key in combined_keys:
+            # Expect each combined key as a key in the returned dict
+            self.assertIn(key, ax_dict.keys())
+
+            # Each should be a matplotlib axis object
+            self.assertIsInstance(ax_dict[key], plt.Axes)
+
     def tearDown(self):
         del self.adata
 

```

Comment:
@ruiheesi, should we add another test to check that we actually calculated the spatial interaction for the stratified subsets of the original adata? Your ideas to make that test are welcomed.

Reply:
Hi George, I don't think it's necessary as we already tested the subset with key checking, adding a check to see if subset is perform correct will actually be testing the 'adata.obs['concatenated_obs'].unique()' function.